samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -56,10 +72,59 @@ def _replace_property(property_key, property_value, resource, logical_id):
 
         """
         if property_key and property_value:
-            resource.get(PROPERTIES_KEY, {})[property_key] = property_value
+            nested_keys = property_key.split(".")
+            target_dict = resource.get(PROPERTIES_KEY, {})
+            while len(nested_keys) > 1:
+                key = nested_keys.pop(0)
+                target_dict[key] = {}
+                target_dict = target_dict[key]
+            target_dict[nested_keys[0]] = property_value
         elif property_key or property_value:
             LOG.info(
                 "WARNING: Ignoring Metadata for Resource %s. Metadata contains only aws:asset:path or "
                 "aws:assert:property but not both",
                 logical_id,
             )
+
+    @staticmethod
+    def _extract_image_asset_metadata(metadata):
```

Comment:
You should have tests that validate windows works as expected. Are Paths provided by the Metadata (aka CDK here) always paths that match the system? Meaning should we expect windows paths on windows or unix paths on windows?

Reply:
I did some messing around with CDK on Windows and it looks like `aws:asset:path` will return the system-matching paths, whereas `aws:asset:dockerfile-path` is always a unix-style relative path. I added a test for this Windows case.